diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 39b96803c7..31883bce56 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -822,7 +822,7 @@
           'conditions': [
             ['v8_enable_webassembly==1', {
               'conditions': [
-                ['OS=="mac" or (_toolset=="host" and host_arch=="x64" and OS=="linux")', {
+                ['OS=="mac" or OS=="ios" or (_toolset=="host" and host_arch=="x64" and (host_os=="linux" or host_os=="mac"))', {
                   'sources': [
                     '<(V8_ROOT)/src/trap-handler/handler-inside-posix.cc',
                     '<(V8_ROOT)/src/trap-handler/handler-outside-posix.cc',
@@ -836,7 +836,7 @@
                   ],
                 }],
                 # TODO(targos): Replace False with OS=="win" if handler-outside-simulator.cc becomes compatible with MSVC.
-                ['_toolset=="host" and host_arch=="x64" and (OS=="linux" or OS=="mac" or False)', {
+                ['_toolset=="host" and host_arch=="x64" and (host_os=="linux" or host_os=="mac" or False)', {
                   'sources': [
                     '<(V8_ROOT)/src/trap-handler/handler-outside-simulator.cc',
                   ],
