diff --git a/node.gyp b/node.gyp
index 63f60a924b..d1316050b3 100644
--- a/node.gyp
+++ b/node.gyp
@@ -79,7 +79,7 @@
       [ 'OS=="win" and '
         'node_use_openssl=="true" and '
         'node_shared_openssl=="false"', {
-        'use_openssl_def%': 1,
+        'use_openssl_def%': 0,
       }, {
         'use_openssl_def%': 0,
       }],
@@ -621,6 +621,7 @@
             'Dbghelp',
             'Psapi',
             'Ws2_32',
+            'Winmm',
           ],
         }],
         [ 'node_use_etw=="true"', {
diff --git a/vcbuild.bat b/vcbuild.bat
index b5c5a53dd7..d2410b4319 100644
--- a/vcbuild.bat
+++ b/vcbuild.bat
@@ -68,7 +68,7 @@ set no_cctest=
 set cctest=
 set openssl_no_asm=
 set doc=
-set extra_msbuild_args=
+set extra_msbuild_args=/p:PlatformToolset=v141
 set exit_code=0
 
 :next-arg
