diff --git a/node.gyp b/node.gyp
index 74e9f9cc..0d52c3b0 100644
--- a/node.gyp
+++ b/node.gyp
@@ -80,7 +80,7 @@
       [ 'OS=="win" and '
         'node_use_openssl=="true" and '
         'node_shared_openssl=="false"', {
-        'use_openssl_def%': 1,
+        'use_openssl_def%': 0,
       }, {
         'use_openssl_def%': 0,
       }],
@@ -641,6 +641,7 @@
             'Dbghelp',
             'Psapi',
             'Ws2_32',
+            'Winmm',
           ],
         }],
         [ 'node_use_etw=="true"', {
