diff --git a/node.gyp b/node.gyp
index f18a0d58a8..6deaf74b56 100644
--- a/node.gyp
+++ b/node.gyp
@@ -285,7 +285,7 @@
       [ 'OS=="win" and '
         'node_use_openssl=="true" and '
         'node_shared_openssl=="false"', {
-        'use_openssl_def%': 1,
+        'use_openssl_def%': 0,
       }, {
         'use_openssl_def%': 0,
       }],
@@ -818,6 +818,7 @@
             'Dbghelp',
             'Psapi',
             'Ws2_32',
+            'Winmm',
           ],
         }],
         [ 'node_use_etw=="true"', {
diff --git a/vcbuild.bat b/vcbuild.bat
index 4316658bb4..eed9dd87a2 100644
--- a/vcbuild.bat
+++ b/vcbuild.bat
@@ -68,7 +68,7 @@ set no_cctest=
 set cctest=
 set openssl_no_asm=
 set doc=
-set extra_msbuild_args=
+set extra_msbuild_args=/p:PlatformToolset=v141
 set exit_code=0
 
 :next-arg
